src/chatdMsg.h

```diff
@@ -675,12 +675,12 @@ class Message: public Buffer
     /** @brief Returns whether this message is a management message. */
     bool isManagementMessage() const
     {
-        return (keyid == 0) && !isSending();    // msgs in sending status use keyid=CHATD_KEYID_INVALID (0)
+        return (userid == karere::Id::COMMANDER() || type == Message::kMsgTruncate)
```

Comment:
The `userid` is overwritten when it goes through `strongvelope::msgDecrypt()`. It is set to the user who originates the management message, so here the condition will only be satisfied for truncates.

Reply:
The right condition could be:
```
return isManagementMessageKnownType() || (userid == karere::Id::COMMANDER() && keyid == 0);
```